cli/popper/commands/cmd_run.py

```diff
@@ -34,10 +34,25 @@
     required=False,
     default=None
 )
+@click.option(
+    '--recursive',
+    help='Run any main.workflow found recursively from current path. ',
+    required=False,
+    is_flag=True
+)
 @pass_context
-def cli(ctx, action, wfile, workspace, reuse):
+def cli(ctx, action, wfile, workspace, reuse, recursive):
     """Executes one or more pipelines and reports on their status.
     """
+
+    if recursive:
+        for root, dirs, files in os.walk('.'):
+            for file in files:
+                if file.endswith('.workflow'):
+                    wfile = os.path.join(root, file)
+                    wfile = os.path.abspath(wfile)
+                    pu.info("Found and running workflow at "+wfile+"\n")
```

Comment:
This will only run the last `.workflow` file found in the folder hierarchy. Instead we would like to create a `Workflow` object for each of the `.workflow` files found. This would be solved by indenting lines 56-71 to the innermost level of the loop.

Reply:
If 56-71 are shifted, won't the rest of code be dependent on `recursive` variable? Used an `else` at line 56 instead so that original functionality is maintained.